raiden/network/transport/matrix/rtc/web_rtc.py

```diff
@@ -0,0 +1,217 @@
+        self._handle_sdp_callback = handle_sdp_callback
+        self.address_to_rtc_partners: Dict[Address, RTCPartner] = dict()
+        self.coroutines: List[Coroutine] = list()
+
+    def get_rtc_partner(self, partner_address: Address) -> RTCPartner:
+        if partner_address not in self.address_to_rtc_partners:
+            self.address_to_rtc_partners[partner_address] = RTCPartner(
+                partner_address, RTCPeerConnection()
+            )
+        return self.address_to_rtc_partners[partner_address]
+
+    def has_ready_channel(self, partner_address: Address) -> bool:
+        if partner_address not in self.address_to_rtc_partners:
+            return False
+        channel = self.address_to_rtc_partners[partner_address].channel
+        if channel is None:
+            return False
+        if channel.readyState == "open":
+            return True
+        return False
+
+    def async_create_channel(self, partner_address: Address) -> None:
+        assert self.node_address, "Transport is not started yet but tried to create rtc channel"
+
+        rtc_partner = self.get_rtc_partner(partner_address)
+        spawn_coroutine(
+            coroutine=create_channel(
+                rtc_partner, self.node_address, self._handle_message_callback
+            ),
+            callback=self._handle_sdp_callback,
+            partner_address=partner_address,
+        )
+
+    def async_set_remote_description(
+        self, partner_address: Address, description: Dict[str, str]
+    ) -> None:
+        assert self.node_address, "Transport is not started yet but tried to set candidates"
+        rtc_partner = self.get_rtc_partner(partner_address)
+        spawn_coroutine(
+            coroutine=set_remote_description(
+                rtc_partner=rtc_partner,
+                node_address=self.node_address,
+                description=description,
+                handle_message_callback=self._handle_message_callback,
+            ),
+            callback=self._handle_sdp_callback,
+            partner_address=partner_address,
+        )
+
+    def stop(self) -> None:
+        if self.node_address is None:
+            return
+
+        log.debug("Gracefully closing RTC channels", node=to_checksum_address(self.node_address))
+
+        for rtc_partner in self.address_to_rtc_partners.values():
+            if rtc_partner.channel:
+                rtc_partner.channel.close()
+
+
+async def create_channel(
+    rtc_partner: RTCPartner, node_address: Address, handle_message_callback: Callable
+) -> RTCSessionDescription:
+
+    pc = rtc_partner.peer_connection
+    rtc_partner.create_channel(node_address)
+
+    offer = await pc.createOffer()
+    await pc.setLocalDescription(offer)
+
+    msg = "Channel must be created already"
+    assert rtc_partner.channel is not None, msg
+
+    @rtc_partner.channel.on("message")
+    async def on_message(message: Dict[str, bytes]) -> None:  # pylint: disable=unused-variable
```

Comment:
This will create a new handler for each channel,m can we use one instead and just link it here?

Reply:
I could also call it in an undecorated form `channel.on("message", callback)` but the problem is that I use outer scope variables which differ in each channel (`partner_address`). Do you have an idea how to solve this problem? We need the partner address to identify the partner in the transport. 
Unfortunately, the code base of aiortc does not look like the possibility to call the callback with additional args or kwargs

Reply:
@ulope Any smart way to do this?

Reply:
Yeah, `functools.partial()` is the way to go in cases like this.

E.g.:
```
def some_handler(external_arg_needed, callback_arg):
    pass

def call_callback(callback):
    callback("callback_arg")

def make_callback():
    cb = partial(some_handler, "first arg value")
    # ^ This makes a new callable that has `"first arg value"` "baked in" when being called. 
    call_callback(cb)
```

Reply:
actually, as just discussed with @ulope, this still creates a new handler for each channel, but gives the chance to define the function globally instead of inline. Performancewise this doesnt change anything